### PR TITLE
Added State retention on rotation functionality for all the activities

### LINKED ISSUE #2229: Two bottom navigation tabs appear to be selected at once in Editing Activity
**Actual Behaviour**

Two bottom navigation tabs appear to be selected at once in Editing Activity while navigating between different tabs.

**Expected Behaviour**

Only the current tab the user has selected should be shown as the selected one.

**Steps to reproduce it**
1) Select an image and go to the editing activity.
2) The filter tab is shown as selected one. Now switch to landscape mode.
3) Press any other tab apart from filter(like stickers).
4) Both filter and stickers are shown as selected tabs. 

**Screenshots of the issue**

6ccf422a-cab9-4e1b-b829-c466c05ed108

696185b9-5ac0-400d-bcea-744893bda503

**Would you like to work on the issue?**
Yes

### PR DESCRIPTION
**Pull request by codepoet2017390**
Fixed #2229 #2175 #2192 

Changes: 

This pull request provides state retention when **Edit is Saved** for all the Activities which include
- Crop
- Filters
- Enhance
- Rotate
-  Stickers
- Add text
- Paint

except one activity
- Frames 

Im still working on Frames activity and it was already referenced in Issue #2230    

This pull request also provides state retention when **Edit is Not Saved** for all the Activities which include
- Crop
- Filters
- Enhance
-  Stickers
- Add text
- Paint

except two activities
- Rotate (Has a different architecture as compared to other activities, Im still working on it)
- Frames

Basically i have used Bundle objects to save instances or state defining parameters to save the state and then display in the new orientation
  
Screenshots of the change: 

**Stickers Activity**

stickers

**Crop Activity**

crop

**enhance**

enhance

**Paint**

paint

**Filter**

**Rotate**

rotate

**Add text**

add text

### CODE REVIEW

**Comment by abishekvashok** [comment]
Conflicts

**Comment by codepoet2017390** [comment]
@abishekvashok  updated

File: app/src/main/java/org/fossasia/phimpme/editor/view/RotateImageView.java [hunk 1]
```diff
@@ -8,16 +8,20 @@
 import android.graphics.Rect;
 import android.graphics.RectF;
 import android.util.AttributeSet;
+import android.util.Log;
 import android.view.View;
 
+import org.fossasia.phimpme.editor.EditImageActivity;
```
[the remark this answers is not in the document]
Comment (abishekvashok) [reply app/src/main/java/org/fossasia/phimpme/editor/view/RotateImageView.java]: Ununsed imports

**Comment by codepoet2017390** [comment]
@abishekvashok  @sauravvishal8797  please review

**Comment by abishekvashok** [comment]
Closing as stale
